aten/src/ATen/test/mps_test_allocator.cpp

```diff
@@ -35,5 +38,6 @@ TEST(MPSAllocator, MPSAllocatorCallbacks) {
         }
         replay_buffer.push_back(new_value);
     }
+    torch::mps::synchronize();
```

Comment:
@kulinseth Is there doc about what this op does? Do we actually need to call it here?
If this is just to check the function is available, we should have a comment saying so.

Reply:
@albanD , the synchronize is documented [here ](https://github.com/pytorch/pytorch/pull/96668/files#diff-46701bb5876c77c75fc9577322d64f2f41daa5112d8ac83c3da9641679b1d854R17). It's just a way for MPS to synchronize with CPU, similar to CUDA synchronize function. Do you think there is value to add this in the MPS docs ?

Reply:
Yes, the `torch.mps.synchronize()` is documented [in here](https://github.com/pytorch/pytorch/pull/94417/files#diff-3a9cc21de8cd29d75bfce4d86e4ebdbe8a68fb937b8a7b972dcc7f73d7e710d1R17) and compiled via `mps.rst`. So it's documentation should be available on on torch-2.0.

Reply:
I added a comment for that line anyway. Thanks.

Reply:
Yeah I think it is an ok description. But my guess is that people may wonder when they need to call this or not.
Not that the cuda version does much better. But I think we should mention that we don't expect the user to ever call this unless they're doing benchmarking on need a full sync with the GPU for some reason.